pkg/sql/sqlbase/schema_desc.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sqlbase
+
+// SchemaDescriptorInterface will eventually be called dbdesc.Descriptor.
+// It is implemented by ImmutableSchemaDescriptor.
+type SchemaDescriptorInterface interface {
+	BaseDescriptorInterface
+	SchemaDesc() *SchemaDescriptor
+}
+
+var _ SchemaDescriptorInterface = (*ImmutableSchemaDescriptor)(nil)
+
+// ImmutableSchemaDescriptor wraps a Schema descriptor and provides methods
+// on it.
+type ImmutableSchemaDescriptor struct {
+	SchemaDescriptor
+}
+
+// MutableSchemaDescriptor is a mutable reference to a SchemaDescriptor.
+//
+// TODO(ajwerner): Today this isn't actually ever mutated but rather exists for
```

Comment:
I don't think it's that bad, especially if we embed the immutable one within the mutable one.

Reply:
Moved to a `Note` to clarify the intention to keep changing the structure here.